### PR TITLE
Dmp 4137 armrpo integrate save background search

### PR DESCRIPTION
**Pull request by karen-hedges**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DMP-4137

### Change description ###

Added functionality to call the ARM RPO SaveBackgroundSearch api

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/darts/arm/rpo/impl/ArmRpoApiImpl.java [hunk 1]
```diff
@@ -236,7 +231,30 @@ public ArmAsyncSearchResponse addAsyncSearch(String bearerToken, Integer executi
...
+            saveBackgroundSearchResponse = armRpoClient.saveBackgroundSearch(bearerToken, saveBackgroundSearchRequest);
+        } catch (FeignException e) {
+            // this ensures the full error body containing the ARM error detail is logged rather than a truncated version
+            log.error(errorMessage.append("Unable to save background search").toString() + " {}", e.contentUTF8());
+            throw handleFailureAndCreateException(errorMessage.toString(), armRpoExecutionDetailEntity, userAccount);
+        }
+
+        if (isNull(saveBackgroundSearchResponse)
+            || isNull(saveBackgroundSearchResponse.getStatus())
+            || !saveBackgroundSearchResponse.getStatus().equals(ARM_RPO_SUCCESS_RESPONSE_STATUS)) {
```
Comment (chrisbellingham-hmcts) [inline src/main/java/uk/gov/hmcts/darts/arm/rpo/impl/ArmRpoApiImpl.java:252]: The other code we have in this class (that calls the other ARM endpoints) is happy with any 2xx "success" response (any non-2xx response will throw the FeignException).

So do we need to explicitly check that the response code in the response body is 200 here? Might not be a bad idea, but perhaps we should either do it consistently across all the ARM calls, or not at all?

File: src/main/java/uk/gov/hmcts/darts/arm/rpo/impl/ArmRpoApiImpl.java [hunk 2]
```diff
@@ -277,6 +295,13 @@ private ArmRpoException handleFailureAndCreateException(String message,
         return new ArmRpoException(message);
     }
 
+    private static StorageAccountRequest createStorageAccountRequest() {
```
Comment (chrisbellingham-hmcts) [inline src/main/java/uk/gov/hmcts/darts/arm/rpo/impl/ArmRpoApiImpl.java:298]: Can be non-static for consistency with the other private methods.

File: src/test/java/uk/gov/hmcts/darts/arm/rpo/impl/ArmRpoApiSaveBackgroundSearchTest.java [hunk 3]
```diff
@@ -1,22 +1,149 @@
...
+        // when
+        armRpoApi.saveBackgroundSearch("token", 1, "searchName", userAccount);
+
+        // then
+        verify(armRpoService).updateArmRpoStateAndStatus(any(),
+                                                         eq(ARM_RPO_HELPER_MOCKS.getSaveBackgroundSearchRpoState()),
+                                                         eq(ARM_RPO_HELPER_MOCKS.getInProgressRpoStatus()),
+                                                         any());
+        verify(armRpoService).updateArmRpoStatus(any(), eq(ARM_RPO_HELPER_MOCKS.getCompletedRpoStatus()), any());
+    }
```
Comment (chrisbellingham-hmcts) [inline src/test/java/uk/gov/hmcts/darts/arm/rpo/impl/ArmRpoApiSaveBackgroundSearchTest.java:79]: Consider adding a `verifyNoMoreInteractions(armRpoService);` at the end to verify there is no unexpected 3rd interaction after moving the status to complete (same comment applies to all tests here).

**Review by chrisbellingham-hmcts — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/darts/arm/rpo/impl/ArmRpoApiImpl.java (hunk 2 above)
Comment (karen-hedges) [reply src/main/java/uk/gov/hmcts/darts/arm/rpo/impl/ArmRpoApiImpl.java:298]: Good spot, I don't know why when I refactor code it automatically adds static each time

**Review by chrisbellingham-hmcts — APPROVED** [review]

**Review by hemantasharma1129 — APPROVED** [review]

**Pull request merged**